src/nomad_simulations/physical_property.py

```diff
@@ -0,0 +1,231 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Any, Optional
+
+from nomad import utils
+from nomad.datamodel.data import ArchiveSection
+from nomad.metainfo import (
+    Quantity,
+    SubSection,
+    SectionProxy,
+    Reference,
+    Section,
+    Context,
+    MEnum,
+)
+from nomad.metainfo.metainfo import DirectQuantity, Dimension, _placeholder_quantity
+from nomad.datamodel.metainfo.basesections import Entity
+
+from .variables import Variables
+from .numerical_settings import SelfConsistency
+
+
+# We add `logger` for the `PhysicalProperty.variables_shape` method
+logger = utils.get_logger(__name__)
+
+
+class PhysicalProperty(ArchiveSection):
+    """
+    A base section used to define the physical properties obtained in a simulation, experiment, or in a post-processing
+    analysis. Besides its `value`, it also contains:
+        - Quantities that are string identifiers: `source`, `type`, `label`
+        - Quantities for calculating the shape of the physical property: `shape`, `variables`
+        - Quantities for referencing and establishing the character of a physical property: `outputs_ref`, `entity_ref`, `self_consistent_ref`,
+        `is_derived`, `is_scf_converged`
+    """
+
+    name = Quantity(
+        type=str,
+        description="""
+        Name of the physical property. Example: `'ElectronicBandGap'`.
+        """,
+    )
+
+    source = Quantity(
+        type=MEnum('simulation', 'measurement', 'analysis'),
+        default='simulation',
+        description="""
+        Source of the physical property. Example: an `ElectronicBandGap` can be obtained from a `'simulation'`
+        or in an `'measurement'`.
+        """,
+    )
+
+    type = Quantity(
+        type=str,
+        description="""
+        Type categorization of the physical property. Example: an `ElectronicBandGap` can be `'direct'`
+        or `'indirect'`.
+        """,
+        # ! add more examples in the description
+    )
```

Comment:
Yes, I agree, but the problem is that we should try to standarize a bit the strings that serve as identifiers / classifiers in the metainfo (as @JFRudzinski pointed out, there is a lot of inconsistent naming). Here, our goal should be "let's try to define the minimal number of them which are useful". 

Now, I'd argue that `type` is useful because it is easier to look if a band gap has some string saying "direct" or "indirect", rather than looking at the momentum. But there might be other situations in which someone wants to redefine it a bit for convenience. Hence the comment on "add more examples".

Similarly, for me, `label` would be very useful for identifying `method_label`, i.e.,  if a property has been derived from one or another methodology. Again, you can argue "but you can look into the references and you will find the same information", and yes, but a simple string does not hurt. Instead of defining `method_label`, I think I could be using `label`, as well as other people could repurpose it in their specific properties. And because this is pretty much case-dependent, I see little to no overlapping on conflicting label naming.

> Let me play devil's advocate and ask you: why stop with type and label? There are also kind, category, classification, etc.

I answered a bit above, but for me `kind` and `type` serve the same purpose. Similarly, `category` would help us further classify, but I will keep it out for now (we might not even want to define it, but perhaps there is some interesting connection with the FAIRmat taxonomy). I am not sure what you meant with `classification`, but I am guessing that it could serve the same purpose than `category`?

_____

Now, if in the future we find use-cases where none of this holds, then I would suggest to revisit it and come with a better idea of how to handle easy string identifiers. **But, we need more examples**.